capabilityreceiver/pkg/chplmapper/capabilityinfo.go

```diff
@@ -62,16 +62,44 @@ func MatchEndpointToProduct(ctx context.Context, ep *endpointmanager.FHIREndpoin
 	if err != nil {
 		return errors.Wrap(err, "error matching the capability statement to a CHPL product")
 	}
-	chplID := chplProductNameVersion[softwareName][softwareVersion]
 
-	healthITProductID, err := store.GetHealthITProductIDByCHPLID(ctx, chplID)
-	// No errors thrown means a healthit product with CHPLID was found and can be set on ep
-	if err == nil {
-		healthITMapID, err := store.AddHealthITProductMap(ctx, ep.HealthITProductID, healthITProductID)
-		if err != nil {
-			return err
+	chplIDArr := []string{}
+
+	chplIDMatchFile := chplProductNameVersion[softwareName][softwareVersion]
+
+	if len(chplIDMatchFile) != 0 {
+		chplIDArr = append(chplIDArr, chplIDMatchFile)
+	}
+
+	healthITProductsArr, err := store.GetActiveHealthITProductsUsingName(ctx, softwareName)
+	if err != nil {
+		return err
+	}
+
+	for _, healthITProduct := range healthITProductsArr {
+		if len(softwareVersion) == 0 {
+			if !arrContains(chplIDArr, healthITProduct.CHPLID) {
```

Comment:
We already have a StringArrayContains function in the endpoint/helpers file, so i think we could use that here and below instead of creating a new function!